src/overlay025/poketch_system.c

```diff
@@ -489,294 +489,287 @@ static void PoketchSystem_UnloadApp(PoketchSystem *poketchSys)
 }
 
-static void ov25_0225420C (PoketchSystem *poketchSys)
+static void PoketchSystem_ShutdownApp(PoketchSystem *poketchSys)
 {
-    GF_ASSERT(poketchSys->unk_02 == 2);
+    GF_ASSERT(poketchSys->appState == POKETCH_APP_STATE_UPDATE);
 
-    poketchSys->unk_40(poketchSys->unk_24);
-    poketchSys->unk_02 = 3;
+    poketchSys->currAppShutdown(poketchSys->unk_24);
+    poketchSys->appState = POKETCH_APP_STATE_SHUTDOWN;
 }
 
-static BOOL ov25_02254228 (PoketchSystem *poketchSys)
+static BOOL PoketchSystem_IsAppShutdown(PoketchSystem *poketchSys)
 {
-    if (poketchSys->unk_02 == 0) {
+    if (poketchSys->appState == POKETCH_APP_STATE_NONE) {
         return TRUE;
     }
 
     return FALSE;
 }
 
-void ov25_02254238 (UnkFuncPtr_ov25_02254238 param0, UnkFuncPtr_ov25_02254238_1 param1)
+void PoketchSystem_SetAppFunctions(PoketchAppInitFunction appInit, PoketchAppShutdownFunction appShutdown)
 {
-    PoketchSystem * v0 = ov25_02253CE0();
+    PoketchSystem *poketchSys = PoketchSystem_GetFromFieldSystem();
 
-    v0->unk_3C = param0;
-    v0->unk_40 = param1;
-    v0->unk_44 = NULL;
+    poketchSys->currAppInit = appInit;
+    poketchSys->currAppShutdown = appShutdown;
+    poketchSys->currAppSave = NULL;
 }
 
-void ov25_0225424C (PoketchSystem *poketchSys)
+void PoketchSystem_NotifyAppLoaded(PoketchSystem *poketchSys)
 {
-    GF_ASSERT(poketchSys->unk_02 == 1);
-    poketchSys->unk_02 = 2;
+    GF_ASSERT(poketchSys->appState == POKETCH_APP_STATE_INIT);
+    poketchSys->appState = POKETCH_APP_STATE_UPDATE;
 }
 
-void ov25_02254260 (PoketchSystem *poketchSys)
+void PoketchSystem_NotifyAppUnloaded(PoketchSystem *poketchSys)
 {
-    GF_ASSERT(poketchSys->unk_02 == 3);
-    poketchSys->unk_02 = 0;
+    GF_ASSERT(poketchSys->appState == POKETCH_APP_STATE_SHUTDOWN);
+    poketchSys->appState = POKETCH_APP_STATE_NONE;
 }
 
-void ov25_02254274 (UnkFuncPtr_ov25_02254274 param0, void * param1)
+void PoketchSystem_SetSaveFunction(PoketchAppSaveFunction saveFunction, void *saveData)
 {
-    PoketchSystem * v0 = ov25_02253CE0();
+    PoketchSystem *poketchSys = PoketchSystem_GetFromFieldSystem();
 
-    v0->unk_44 = param0;
-    v0->unk_48 = param1;
+    poketchSys->currAppSave = saveFunction;
+    poketchSys->appSaveData = saveData;
 }
 
-static BOOL ov25_02254284 (PoketchSystem *poketchSys)
-{
-    static const TouchScreenHitTable v0[] = {
-        {4 * 8, 12 * 8, 28 * 8, 255},
-        {12 * 8, 20 * 8, 28 * 8, 255},
-        {16, 175, 16, 207}
-    };
+static const TouchScreenHitTable sMainPoketchButtons[] = {
+    {POKETCH_BUTTON_TOP_MINY,    POKETCH_BUTTON_TOP_MAXY,    POKETCH_BUTTON_TOP_MINX,    POKETCH_BUTTON_TOP_MAXX},      // Top button
+    {POKETCH_BUTTON_BOTTOM_MINY, POKETCH_BUTTON_BOTTOM_MAXY, POKETCH_BUTTON_BOTTOM_MINX, POKETCH_BUTTON_BOTTOM_MAXX},   // Bottom Button
+    {POKETCH_SCREEN_MINY,        POKETCH_SCREEN_MAXY,        POKETCH_SCREEN_MINX,        POKETCH_SCREEN_MAXX}           // Screen
+};
```

Comment:
[polish] I would style this as:

```c
enum {
    POKETCH_BUTTON_TOP = 0,
    POKETCH_BUTTON_BOTTOM,
    POKETCH_BUTTON_SCREEN,
};

static const TouchScreenHitTable sMainPoketchButtons[] = {
    [POKETCH_BUTTON_TOP]    = {POKETCH_BUTTON_TOP_MINY,    POKETCH_BUTTON_TOP_MAXY,    POKETCH_BUTTON_TOP_MINX,    POKETCH_BUTTON_TOP_MAXX},
    [POKETCH_BUTTON_BOTTOM] = {POKETCH_BUTTON_BOTTOM_MINY, POKETCH_BUTTON_BOTTOM_MAXY, POKETCH_BUTTON_BOTTOM_MINX, POKETCH_BUTTON_BOTTOM_MAXX},
    [POKETCH_BUTTON_SCREEN] = {POKETCH_SCREEN_MINY,        POKETCH_SCREEN_MAXY,        POKETCH_SCREEN_MINX,        POKETCH_SCREEN_MAXX},
};
```

It is a bit more verbose, but leverages code syntax to both enforce a match and document what each element of this table corresponds to. You can also use the enum later for more semantic access to the table, if desired.